datafusion/sqllogictest/test_files/scalar.slt

```diff
@@ -1779,6 +1779,18 @@ SELECT COALESCE(NULL, 'test')
 ----
 test
 
+
+statement ok
+create table test1 as values (arrow_cast('foo', 'Dictionary(Int32, Utf8)')), (null);
+
+
+query ?
+select  coalesce(column1, 'none_set') from test1;
+----
+foo
+none_set
+
```

Comment:
Can we add a few more tests here?

1. The alternate (cast utf8 to dictionary):

```sql
select coalesce('none_set', column1) from test1;
select coalesce(NULL, 'none_set', column1) from test1;
```

3. A test on non string dictionaries, like this:
```sql
select coalesce(null, 34, arrow_cast(123, 'Dictionary(Int32, Int8)'));
select coalesce(null, arrow_cast(123, 'Dictionary(Int32, Int8)'), 34);
```
3. Also clear the table (e.g. `drop table test1`)?

Reply:
sure